src/App.tsx

```diff
@@ -1,44 +1,156 @@
+import { queryFirebase, getLocationDict } from "./firebase";
+import {
+    getSubmitPackingUrl,
+    packingStatusUrl,
+    getLogsUrl,
+    JobStatus,
+} from "./constants/awsBatch";
+import {
+    FIRESTORE_COLLECTIONS
+} from "./constants/firebaseConstants";
+import {
+    AWSBatchJobsResponse,
+    CloudWatchLogsResponse,
+    StringDict,
+} from "./types";
 
 function App() {
-    const [count, setCount] = useState(0);
-    const [text, setText] = useState("");
+    const [recipes, setRecipes] = useState<StringDict>({});
+    const [configs, setConfigs] = useState<StringDict>({});
+    const [selectedRecipe, setSelectedRecipe] = useState("");
+    const [selectedConfig, setSelectedConfig] = useState("");
+    const [jobId, setJobId] = useState("");
+    const [jobStatus, setJobStatus] = useState("");
+    const [logStreamName, setLogStreamName] = useState("");
+    const [jobLogs, setJobLogs] = useState<string[]>([]);
+
+    const submitRecipe = async () => {
+        const url = getSubmitPackingUrl(selectedRecipe, selectedConfig);
+        const request: RequestInfo = new Request(url, {
+            method: "POST",
+        });
+        const response = await fetch(request);
+        const data = await response.json();
+        setJobId(data.jobId);
+        return data.jobId;
+    };
+
+    const getRecipes = async () => {
+        const recipeDict = await getLocationDict(FIRESTORE_COLLECTIONS.EXAMPLE_RECIPES);
+        return recipeDict;
+    };
+
+    useEffect(() => {
+        const fetchRecipes = async () => {
+            const recipeDict = await getRecipes();
+            setRecipes(recipeDict);
+        };
+        fetchRecipes();
+    }, []);
+
+    const getConfigs = async () => {
+        const configDict = await getLocationDict(FIRESTORE_COLLECTIONS.CONFIGS);
+        return configDict;
+    };
+
+    useEffect(() => {
+        const fetchConfigs = async () => {
+            const configDict = await getConfigs();
+            setConfigs(configDict);
+        };
+        fetchConfigs();
+    }, []);
+
+    const checkStatus = async (jobIdFromSubmit: string) => {
+        const id = jobIdFromSubmit || jobId;
+        const url = packingStatusUrl(id);
+        const request: RequestInfo = new Request(url, {
+            method: "GET",
+        });
+        let localJobStatus = "";
+        while (localJobStatus !== JobStatus.SUCCEEDED && localJobStatus !== JobStatus.FAILED) {
+            const response = await fetch(request);
+            const data: AWSBatchJobsResponse = await response.json();
+            if (localJobStatus !== data.jobs[0].status) {
+                localJobStatus = data.jobs[0].status;
+                setJobStatus(data.jobs[0].status);
+            }
+            setLogStreamName(data.jobs[0].container.logStreamName);
+        }
+    };
+
+    const fetchResultUrl = async () => {
+        const url = await queryFirebase(jobId);
+        window.open("https://simularium.allencell.org/viewer?trajUrl=" + url);
```

Comment:
This url string has been moved to constants in a separate [pr](https://github.com/mesoscope/cellpack-client/pull/7)